salt/metalk8s/kubeadm/init/preflight/mandatory.sls

```diff
@@ -0,0 +1,92 @@
+{%- from "metalk8s/map.jinja" import defaults with context %}
+{%- from "metalk8s/map.jinja" import kubeadm_preflight with context %}
+{%- from "metalk8s/map.jinja" import repo with context %}
+
+include:
+  - metalk8s.repo
+
+Install mandatory packages:
+  pkg.installed:
+    - pkgs: {{ kubeadm_preflight.mandatory.packages }}
+
+{%- for service in kubeadm_preflight.mandatory.services.running %}
+Enable {{ service  }} service:
+  service.running:
+    - name: {{ service  }}
+    - enable: true
+{%- endfor %}
+
+# required to set sysctl net.bridge.bridge-nf-call-iptables
+Add the module br_netfilter to kernel:
+  kmod.present:
+    - name: br_netfilter
+    - persist: True
+
+{%- for item, value in kubeadm_preflight.mandatory.sysctl_values.items() %}
+Set sysctl {{ item }} value to {{ value }}:
+  sysctl.present:
+    - name: {{ item }}
+    - value: {{ value }}
+    - config: /etc/sysctl.d/60-metalk8s.conf
+    {%- if item in ("net.bridge.bridge-nf-call-ip6tables", "net.bridge.bridge-nf-call-iptables") %}
+    - require:
+      - kmod: Add the module br_netfilter to kernel
+    {%- endif %}
+{%- endfor %}
+
+{%- for swap_device in salt.mount.swaps().keys() %}
+Disable swap device {{ swap_device }}:
+  module.run:
+    - name: mount.swapoff
+    - m_name: {{ swap_device }}
+{%- endfor %}
+
+Prevent swap mount from fstab:
+  file.comment:
+    - name: /etc/fstab
+    - regex: .+\s+.+\s+swap\s+.+
+    - onlyif:
+      # Add a condition to avoid the state failure
+      # in case we don t have swap in /etc/fstab
+      - test -n "$(awk '$2=="swap" {print}' /etc/fstab)"
+
+Install cri-tools package:
+  pkg.installed:
+    - name: cri-tools
+    - fromrepo: {{ repo.kubernetes.name  }}
+    - require:
+      - pkgrepo: Configure Kubernetes repository
+
+Create configuration file for crictl:
+  file.serialize:
+    - name: /etc/crictl.yaml
+    - dataset:
+       runtime-endpoint: unix:///run/containerd/containerd.sock
```

Comment:
You will need to change this after my (new) PR merge or me I will need to change it as we should use the pillar key `kubelet.service.options.container-runtime-endpoint`